app/src/main/java/zegreanu/cristi/geoquiz/CheatActivity.kt

```diff
@@ -8,16 +8,23 @@ import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 
 class CheatActivity : AppCompatActivity() {
+    private val KEY_ANSWER_SHOWN = "answer_shown"
+
     private lateinit var showAnswerButton: Button
     private lateinit var answerTextView: TextView
 
+    private var isAnswerShown: Boolean = false
     private var answerIsTrue: Boolean = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_cheat)
 
         answerIsTrue = intent.getBooleanExtra(EXTRA_ANSWER_IS_TRUE, false)
+        savedInstanceState?.let {
+            isAnswerShown = it.getBoolean(KEY_ANSWER_SHOWN, false)
+            setAnswerShownResult()
```

Comment:
Not sure this is needed :-?. It feels "wrong" to set a result (which is supposedly something that occurs at the end of an operation) when the activity is created.

I think that the first time `setAnswerShownResult` is called, the "result" persists across the recreation of the activity.

Reply:
As discussed, the result is not persistent. We will leave it like this :).